hibernate-core/src/main/antlr/hql-sql.g

```diff
@@ -436,7 +436,7 @@ selectExpr
 	;
 
 count
-    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR ) ) {
+    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR | query) ) {
```

Comment:
@sebersole The concern I have here is truly how portable is this solution across database vendors.  

H2 fails with an exception that states the subquery should only return one row, which when applying a `COUNT DISTINCT` around it makes little sense as it will always be `1`.  

Additionally, MSSQL doesn't support a subquery inside a `COUNT DISTINCT` clause.